### DISCUSSION TITLE
Gesamtproduktion/Tagesproduktion wird abends auf 0 gesetzt

### DISCUSSION THREAD

**Discussion by hoenow** [tbnobody/OpenDTU#1436] [Q&A]
Bis vor kurzem habe ich Hoymiles HM-1500 mit openDTU überwacht
und konnte abends, oder nachts die Produktionswerte des Tages ablesen.
Jetzt habe ich die Wechselrichter gegen HMS-2000 getauscht, weil sie 4 MPP-Tracker haben.
Mit den neuen Wechselrichtern wird die Ertragsanzeige auf 0 gesetzt, wenn die Tagesproduktion endet und der Wechselrichter abschaltet.
Abends die Werte des Tages abzulesen, ist mit den neuen Wechselrichtern nicht mehr möglich.
Bei den HM-1500 blieben die Werte erhalten, bis der Wechselrichter am nächsten Tag neu startete. 

Kennt jemand das Problem oder eine Lösung, wie die Anzeige auch abends möglich ist?

**Comment by Tiese** [comment]
Guten Morgen hoenow,

ist das seit dem neuen Wechselrichter so oder hast du im Zuge des Austauschs auch ein Update von OpenDTU durchgeführt?
Denn in der neuen Version kannst du (unter Einstellungen/Wechselrichter/"Bearbeiten(Stift-Symbol)"/Erweitert) einstellen wann und ob genullt werden soll.

Schöne Grüße
Tiese

**Reply by hoenow** [reply]
Ja, ein Update von openDTU wurde auch ausgeführt.
![Screenshot_openDTU 2023-10-17 131825](https://github.com/tbnobody/OpenDTU/assets/130210757/60197c9c-1a68-401e-9a5f-9c300b012653)
Heute Abend werde ich sehen, ob die Werte erhalten bleiben.
Danke, für den Tipp.....

**Reply by hoenow** [reply]
Die Änderung zeigt keine Wirkung.
Auch mit den geänderten Werten sind die Werte Gesamtertrag Heute, pünktlich beim Abschalten der beiden Wechselrichter auch in der openDTU wieder auf 0 gesetzt.
Ich verwende die aktuelle Firmware v23.10.9
Beide Wechselrichter sind vom Typ HMS-2000.
Beide sind in openDTU gleich eingestellt.

Gibt es noch andere Lösungsansätze?
![Screenshot WR1 2023-10-17 225251](https://github.com/tbnobody/OpenDTU/assets/130210757/f9a126ff-c30e-499d-b443-65c85aa8723f)
![Screenshot WR2 2023-10-17 225514](https://github.com/tbnobody/OpenDTU/assets/130210757/b6277912-0b11-499a-9015-84d151f5eb47)

**Comment by tbnobody** [comment]
Schaue mal hier: https://github.com/tbnobody/OpenDTU/wiki/FAQ#q-why-get-the-daily-yield-values-sometimes-reset-to-zero-at-the-evening

Kannst du ggf. am Abend, wenn der Wert 0 ist ins Event Log schauen? Dort wirst du vmtl. sehen, dass der Inverter irgendwann Abends (neu) gestartet wurde.

**Comment by hoenow** [comment]
Ich habe heute in der Ergebnisanzeige nachgesehen:
18:45 ID1 Inverter start
die gleiche Meldung wie morgens:
07:52 ID1 Inverter start

Dann kann ich mir erklären, dass openDTU einen Reset der Werte macht.
Gibt es eine Möglichkeit, dass per Software abzufangen
oder die Firmware des Wechselrichters zu aktualisieren?

Produktionsjahr | 2022
Produktionswoche | 1
Modell | HMS-2000
Ermittelte max. Leistung | 2.000 W
Bootloader-Version | 0.0.1
Firmware-Version | 1.0.12
Firmware-Erstellungsdatum | 2021-10-25 18:37:00
Hardware-Teilenummer | 269644033
Hardware-Version | 01.00

**Comment by tbnobody** [comment]
```
18:45 ID1 Inverter start
die gleiche Meldung wie morgens:
07:52 ID1 Inverter start
```

Das ist genau das was ich meinte. Bei jedem Inverter Start werden die Erträge genullt. Sprich um 18:45 hat der Wechselrichter aufgrund der Lichtverhältnisse zwischendurch neu gestartet und dadurch wurden die Werte zurückgesetzt. 

Wie schon im FAQ Eintrag geschrieben ist da eine Lösung in Arbeit um das zu umgehen.

**Comment by hoenow** [comment]
Mit Homeassistant habe ich aufgezeichnet, was die beiden Wechselrichter machen, am Ende der Produktion.
Sie schalten nicht nur einfach aus, sondern ständig ein und aus.
![Screenshot WR-8 2023-10-18 195505](https://github.com/tbnobody/OpenDTU/assets/130210757/8cec41db-776e-490b-9128-a264c9d66fa4)

**Comment by tbnobody** [comment]
Ja.... Je nachdem ob es ggf. durch Wolkenänderung usw. nochmal kurz heller wird oder nicht. 

#1268 geht in diese Richtung aber speichert mir noch zu wenig. Wie gesagt, bekanntes Problem....

**Comment by hoenow** [comment]
In der Tat, ein bekanntes Problem, wie ich lese.
https://github.com/tbnobody/OpenDTU/issues/1397

Bleibt die Frage, ob es ein Hardwareproblem im Wechselrichter ist
oder ein Softwarefehler in der Firmware.
Bootloader-Version | 0.0.1
Firmware-Version | 1.0.12
Firmware-Erstellungsdatum | 2021-10-25 18:37:00
Hardware-Teilenummer | 269644033
Hardware-Version | 01.00

Da ich vorher HM-1500 im Einsatz hatte, die diesen Fehler nicht hatten,
gehe ich davon aus, dass Hoymiles das Problem auch kennt und beheben kann.
Die beiden Wechselrichter HMS-2000 habe ich vor einer Woche gekauft
und werde mich an den Lieferanten wenden und den Fehler reklamieren.
Gegebenenfalls die Geräte austauschen lassen.

Ist bekannt, ob es ein Hard- oder Softwareproblem ist?
Besteht die Möglichkeit, eine neuere Firmware in den Wechselrichter einzuspielen?
Gibt es eine Stellungnahme von Hoymiles zu dem Problem?

**Comment by hoenow** [comment]
Hallo tbnobody,

auf der Solarsolutions in Düsseldorf hatte ich einen Techniker von Hoymiles auf das Problem mit dem ständigen
Ein/Ausschalten in der Dämmerung angesprochen und leider keine Lösung angeboten bekommen.
Es gibt eine aktuelle Firmware-Version, 1.0.27 für den HMS-2000.
Ob damit dieses Problem gelöst ist, konnte er nicht sagen.
Ein Firmwareupdate ist leider nur über die original DTU und über die Hoymiles Cloud möglich.

Wäre es möglich, die letzten gültigen Ertragswerte der Panels und der Summe,
im ESP32, in openDTU zu speicher und erst am kommenden Morgen, vor Sonnenaufgang zu resetten?

**Reply by tbnobody** [reply]
Das ist genau das was diese Funktion macht:
![image](https://github.com/tbnobody/OpenDTU/assets/1753222/8c4bd5c4-5a25-4660-ad86-d51a939d5af4)

**Comment by canone2** [comment]
Hallo,habe gerade euren Post gelesen und muss sagen,das ich das selbe Problem habe,nur mit dem Unterschied , ich benutze noch immer den gleichen Hoymiles HM-1500 und seit dem letzten Update 23.12.31 tritt dieses Problem auf. Also müsste es doch ein Softwareproblem sein?Oder sehe ich dies falsch?

Gruß

**Reply by tbnobody** [reply]
Aber du hast oben genannte einstellung aktiviert? Falls das nicht Hilft, Bitte einen Screenshot vom Eventlog einstellen (wenn der Wert abends 0 ist)

**Reply by canone2** [reply]
Habe jetzt auf "Tageskorrektur gestellt" wenn du das meintest.Werde morgen mal testen.Bin noch nicht ganz so firm,wo finde ich den Log?
Gruß

**Reply by tbnobody** [reply]
Dieser Button...
![image](https://github.com/tbnobody/OpenDTU/assets/1753222/b84d2999-2b4d-413a-a7dc-f35737ae3491)

Wenn deine Werte jetzt gerade 0 sind und du Einträge im Eventlog hast sollte der 1. Eintrag ein "Inverter Start" o.ä. sein. Wenn dann die Zeit um 16:00 Uhr o.ä. liegt hat der Inverter zwischenzeitl. wegen zu wenig Licht abgeschaltet und damit den Tageszähler genullt.

Die Funktion die du Gerade aktiviert hast "Tageskorrektur" versucht hier dieses verhalten zu korrigieren und setzt den Tagesertrag erst um Mitternacht zurück. (Mit etwas Magic im Hintergrund)

**Comment by canone2** [comment]
Hier mein Log von heute Abend

Gruß
![20240103_191040](https://github.com/tbnobody/OpenDTU/assets/84138883/c050c45a-9d6a-4d72-b628-b12b87eeb221)
![20240103_191443](https://github.com/tbnobody/OpenDTU/assets/84138883/80ce4822-93d2-4029-affe-d3923f3bebe0)

**Comment by canone2** [comment]
Danke für die Info!Werde testen.
Gruß

> Am 03.01.2024 - 19:19 schrieb tbnobody:
>
>
> Dieser Button...
> image.png (view on web)
> 
> Wenn deine Werte jetzt gerade 0 sind und du Einträge im Eventlog hast sollte der 1. Eintrag ein "Inverter Start" o.ä. sein. Wenn dann die Zeit um 16:00 Uhr o.ä. liegt hat der Inverter zwischenzeitl. wegen zu wenig Licht abgeschaltet und damit den Tageszähler genullt.
> 
> Die Funktion die du Gerade aktiviert hast "Tageskorrektur" versucht hier dieses verhalten zu korrigieren und setzt den Tagesertrag erst um Mitternacht zurück. (Mit etwas Magic im Hintergrund)
> 
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you commented.
> Message ID: ***@***.***>
> 
> 

-------------------------------------------------------------------------------------------------
FreeMail powered by mail.de - MEHR SICHERHEIT, SERIOSITÄT UND KOMFORT

**Comment by hoenow** [comment]
Hallo tbnobody,

die Magie wirkt!
Ich hatte den Punkt in den Wechselrichtereinstellungen vorher nicht gesehen.
Jetzt funktioniert es, wie es soll.
DANKE  !!
![Screenshot 2024-01-03 193325](https://github.com/tbnobody/OpenDTU/assets/130210757/adb475ac-96a9-4155-bd86-d66044e8c2b0)
![Screenshot 2024-01-03 193856](https://github.com/tbnobody/OpenDTU/assets/130210757/58dd89fb-bb63-4693-a654-9674eda4c448)
